src/reducers/versions.tsx

```diff
@@ -80,6 +80,55 @@ export type ExternalVersion = {
   version: string;
 };
 
+type ExternalChange = {
+  content: string;
+  new_line_number: number;
+  old_line_number: number;
+  type: 'normal' | 'delete' | 'insert';
+};
+
+type ExternalHunk = {
+  changes: ExternalChange[];
+  header: string;
+  new_lines: number;
+  new_start: number;
+  old_lines: number;
+  old_start: number;
+};
+
+type ExternalDiff = {
+  hash: string;
+  hunks: ExternalHunk[];
+  is_binary: boolean;
+  lines_added: number;
+  lines_deleted: number;
+  mode: string;
+  old_path: string;
+  parent: string;
+  path: string;
+  size: number;
+};
+
+// An `ExternalVersionFileWithDiff` object is very similar to an
+// `ExternalVersionFile` object: instead of having a `content` field, it has a
+// `diff` field.
+export type ExternalVersionFileWithDiff = Pick<
```

Comment:
I wonder about the naming, though. These new types are called `ExternalVersionFileWithDiff` and `ExternalVersionWithDiff`, which means that the existing types are just called `ExternalVersionFile` and `ExternalVersion`. That means we are making an assumption about what these "default" types contain. Should those be renamed to `ExternalVersionFileWithFile` and `ExternalVersionWithFile` to make it more explicit what each type contains?

Reply:
That would make all the version types more consistent.

Reply:
I believe it should be something `ExternalVersionFileWithContent` and `ExternalVersionWithContent` though.